### PR TITLE
DB connection is created for users table

### PR DESCRIPTION
**Pull request by dev04sa**
@RitwikSingh28  Sir kindly review the PR.
in this PR I have added for if table doesn't exist then create table first then add data to table and if table exist already , then it will insert data to table.(for user only ) .
Here is screenshots are provided : -
Screenshot 2024-04-30 122009
Screenshot 2024-04-30 133442
Screenshot 2024-04-30 133434
Screenshot 2024-04-30 133255
Screenshot 2024-04-30 133241
Screenshot 2024-04-30 132847
Screenshot 2024-04-30 132840
Screenshot 2024-04-30 122015

### CODE REVIEW

File: Backend/controllers/userc.js [hunk 1]
```diff
@@ -1,21 +1,34 @@
...
+    if (!tableCreated) {
+      const createTableQuery = await fs.readFile(filePath, "utf-8");
+      await pool.query(createTableQuery);
+      tableCreated = true;
     }
 
-    // Add user
-    user.push(email);
+    const insertQuery =
+      "INSERT INTO users (Username, Password, UserType) VALUES (?, ?, ?)";
```
Comment (RitwikSingh28) [inline Backend/controllers/userc.js]: What about `UserId`? Firebase Auth will return a unique UserId for each logged in user, that must be stored in the database.

File: Backend/controllers/userc.js [hunk 2]
```diff
@@ -26,13 +39,23 @@ const createUser = async (req, res) => {
 const getUserById = async (req, res) => {
   try {
     const id = req.params.userId;
+    if (id < 0 || isNaN(id)) {
```
Comment (RitwikSingh28) [inline Backend/controllers/userc.js]: The `id` parameter must be a `string`, and not an `int`. Correct it accordingly, and rewrite the check

File: Backend/controllers/userc.js [hunk 3]
```diff
@@ -41,7 +64,27 @@ const getUserById = async (req, res) => {
 
 const updateUser = async (req, res) => {
   try {
-    // Implementation for updating user
+    const userId = req.params.userId;
+    const { Username, Password, UserType } = req.body;
+
+    if (userId < 0 || isNaN(userId)) {
```
Comment (RitwikSingh28) [inline Backend/controllers/userc.js]: Same here. `userId` is a string, and not an `int`

File: Backend/helpers/user.sql [hunk 4]
```diff
@@ -0,0 +1,7 @@
+CREATE TABLE users (
+    UserID INT AUTO_INCREMENT PRIMARY KEY,
```
Comment (RitwikSingh28) [inline Backend/helpers/user.sql]: `UserID` must be a `VARCHAR`, and not an `INT`

**Review by RitwikSingh28 — CHANGES_REQUESTED** [review]
@dev04sa 
Please go through the review comments provided. After having resolved the issues raised, comment on the reviews in the format `Resolved in commit #<commit>`.

File: Backend/controllers/userc.js (hunk 1 above)
Comment (dev04sa) [reply Backend/controllers/userc.js]: Yes that should be here , i will add .

File: Backend/controllers/userc.js (hunk 2 above)
Comment (dev04sa) [reply Backend/controllers/userc.js]: Sure

**Comment by dev04sa** [comment]
> @dev04sa Please go through the review comments provided. After having resolved the issues raised, comment on the reviews in the format `Resolved in commit #<commit>`.

@RitwikSingh28  Sir requested changes done , Resolved in commit #requested changes done

Screenshot 2024-05-07 093841
Screenshot 2024-05-07 093830
Screenshot 2024-05-07 094021

**Review by RitwikSingh28 — APPROVED** [review]
@dev04sa 
All issues resolved, and the merging is good to go! Great work 💯

**Pull request merged**